src/main/resources/META-INF/rewrite/apache-httpclient-5.yml

```diff
@@ -0,0 +1,406 @@
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: org.apache.hc.client5.http.impl.classic.HttpAuthenticator
+      newFullyQualifiedTypeName: org.apache.hc.client5.http.impl.auth.HttpAuthenticator
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: org.apache.hc.client5.http.impl.classic.SystemDefaultCredentialsProvider
+      newFullyQualifiedTypeName: org.apache.hc.client5.http.impl.auth.SystemDefaultCredentialsProvider
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: org.apache.hc.client5.http.impl.classic.DefaultClientConnectionReuseStrategy
+      newFullyQualifiedTypeName: org.apache.hc.client5.http.impl.DefaultClientConnectionReuseStrategy
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: org.apache.hc.client5.http.impl.classic.DefaultRedirectStrategy
+      newFullyQualifiedTypeName: org.apache.hc.client5.http.impl.DefaultRedirectStrategy
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: org.apache.hc.client5.http.impl.classic.RedirectLocations
+      newFullyQualifiedTypeName: org.apache.hc.client5.http.protocol.RedirectLocations
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: org.apache.hc.client5.http.impl.classic.BasicCookieStore
+      newFullyQualifiedTypeName: org.apache.hc.client5.http.cookie.BasicCookieStore
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: org.apache.hc.client5.http.impl.classic.DefaultConnectionKeepAliveStrategy
+      newFullyQualifiedTypeName: org.apache.hc.client5.http.impl.DefaultConnectionKeepAliveStrategy
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: org.apache.hc.client5.http.impl.classic.IdleConnectionEvictor
+      newFullyQualifiedTypeName: org.apache.hc.client5.http.impl.IdleConnectionEvictor
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: org.apache.hc.client5.http.impl.classic.TunnelRefusedException
+      newFullyQualifiedTypeName: org.apache.hc.client5.http.impl.TunnelRefusedException
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: org.apache.hc.client5.http.impl.classic.NoopUserTokenHandler
+      newFullyQualifiedTypeName: org.apache.hc.client5.http.impl.NoopUserTokenHandler
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: org.apache.hc.client5.http.impl.classic.DefaultUserTokenHandler
+      newFullyQualifiedTypeName: org.apache.hc.client5.http.impl.DefaultUserTokenHandler
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: org.apache.hc.client5.http.impl.classic.BasicCredentialsProvider
+      newFullyQualifiedTypeName: org.apache.hc.client5.http.impl.auth.BasicCredentialsProvider
+  # Fixing Depreacted
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: org.apache.hc.client5.http.impl.classic.DefaultHttpClient
+      newFullyQualifiedTypeName: org.apache.hc.client5.http.impl.classic.CloseableHttpClient
```

Comment:
Is the top one still available? Maybe best not to replace just yet in that case.

Reply:
Because if we replace this early, then we might not be able to better target a replacement that introduces try-with-resources later.

Reply:
nope, it's not available anymore... But I'm thinking on not just facing this "deprecated" issues yet, because in general is not as simple as just changing the type, since usually they don't have the same methods available...